src/main/java/com/redhat/devtools/lsp4ij/usages/LSPUsageElementDescriptionProvider.java

```diff
@@ -8,25 +8,79 @@
  * Contributors:
  * Red Hat, Inc. - initial API and implementation
  ******************************************************************************/
+
 package com.redhat.devtools.lsp4ij.usages;
 
+import com.intellij.openapi.editor.Document;
 import com.intellij.openapi.util.NlsSafe;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.openapi.util.text.StringUtil;
 import com.intellij.psi.ElementDescriptionLocation;
 import com.intellij.psi.ElementDescriptionProvider;
 import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.codeStyle.NameUtil;
+import com.intellij.usageView.UsageViewLongNameLocation;
+import com.intellij.usageView.UsageViewShortNameLocation;
+import com.intellij.usageView.UsageViewTypeLocation;
+import com.redhat.devtools.lsp4ij.LSPIJUtils;
 import com.redhat.devtools.lsp4ij.LanguageServerBundle;
+import com.redhat.devtools.lsp4ij.features.semanticTokens.viewProvider.LSPSemanticTokenDocumentationProvider;
+import com.redhat.devtools.lsp4ij.features.semanticTokens.viewProvider.LSPSemanticTokenPsiElement;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 /**
- * {@link ElementDescriptionProvider} implementation used to show "LSP Symbol" as target in the "Find Usages" view.
+ * {@link ElementDescriptionProvider} implementation used to show information about the target in the "Find Usages" view.
  */
 public class LSPUsageElementDescriptionProvider implements ElementDescriptionProvider {
     @Override
-    public @Nullable @NlsSafe String getElementDescription(@NotNull PsiElement element, @NotNull ElementDescriptionLocation location) {
-        if (element instanceof LSPUsageTriggeredPsiElement) {
-            return LanguageServerBundle.message("usage.description");
+    @Nullable
+    @NlsSafe
+    public String getElementDescription(@NotNull PsiElement element, @NotNull ElementDescriptionLocation location) {
+        // For usages
+        if (element instanceof LSPUsageTriggeredPsiElement usageTriggeredPsiElement) {
+            // Element type
+            if (location instanceof UsageViewTypeLocation) {
+                // See if we can get a better type
+                if (usageTriggeredPsiElement.getRealElement() instanceof LSPSemanticTokenPsiElement semanticTokenElement) {
+                    String tokenType = semanticTokenElement.getType();
+                    if (StringUtil.isNotEmpty(tokenType)) {
+                        return StringUtil.join(NameUtil.splitNameIntoWords(tokenType), String::toLowerCase, " ");

```

Comment:
When asked for the usage type, it pulls the token type -- which may be something like `enumMember`; see `SemanticTokenTypes` -- and converts that into words, e.g., `enum member`, which is then capitalized automatically by the caller.